### PR TITLE
#37 adding optional `attributes` parameter;

### PR DESCRIPTION
**Pull request by einsiedlerkrebs**
List of colon separated `KEY:VALUE` pairs of jail attributes (e.g. `devfs_ruleset:3`)

### CODE REVIEW

**Comment by grembo** [comment]
At first glance I see two improvements:
1. Allow turning off the feature by config (as sending in attributes can be dangerous)
2. It suffers from the same quoting problems as other parts of the plugin (see #36)

Maybe you could look into solving those? (for 2. there are various libraries available one could chose from). If you get stuck with 1. we're happy to help.

File: driver/prepare.go [hunk 1]
```diff
@@ -123,6 +123,17 @@ func prepareContainer(cfg *drivers.TaskConfig, taskCfg TaskConfig) (syexec, erro
 		se.argvMountReadOnly = argvMountReadOnly
 	}
 
+    // set attributes
+	if len(taskCfg.Attributes) > 0 {
+		for _, attr := range taskCfg.Attributes {
+			split := strings.Split(attr, ":")
```
Comment (grembo) [inline driver/prepare.go:129]:
```suggestion
			split := strings.SplitN(attr, ":", 2)
```

File: driver/prepare.go [hunk 2]
```diff
@@ -123,6 +123,17 @@ func prepareContainer(cfg *drivers.TaskConfig, taskCfg TaskConfig) (syexec, erro
...
+			attribute := split[0]
+			value := split[1]
+			command := "set-attribute -p " + potName + " -A " + attribute + " -V " + value
```
Comment (grembo) [inline driver/prepare.go:132]: This should contain quoting, so things like `attributes = ["; shutdown -p now:val"]` won't be possible.

**Review by grembo — CHANGES_REQUESTED** [review]

**Comment by grembo** [comment]
@einsiedlerkrebs As we want to release before the end of Q3, I did the most pressing changes myself and merged this now. I'll develop driver config to make this feature opt-in afterwards. Thanks for your contribution!

**Comment by einsiedlerkrebs** [comment]
Wow, that was fast and I am luckily just in time.

Thanks a lot. I will look it over next week.



On 29 September 2023 10:55:34 CEST, grembo ***@***.***> wrote:
>@einsiedlerkrebs As we want to release before the end of Q3, I did the most pressing changes myself and merged this now. I'll develop driver config to make this feature opt-in afterwards. Thanks for your contribution!
>


-- 
encryption? welcome!

gpg: 63e5d348

-----BEGIN PGP PUBLIC KEY BLOCK-----

mQINBFiDKSkBEACjAoZDjZ2GyPrW919ozieezixTX9+k9meyyDOxS+koA3YCdUMk
L7+xwvPgCc/ZY6Ug3EF6+wJxVd+xZcAAGc4Lo8b9Fdrrg/YNbqjitIMA4gatBnEJ
gDO4qrYmwGzA214yINLpSR+Pv6mMv4Y628FltkOnFrVDND3DnqOTLzq8tQHskaLQ
KHgUYNUq93yVjSvquaJgpaJJQbGYyhSehvNLqPRdgC6rvJqAxh4InsclfaGWybxz
cDPDLdUQEefg/kyeY0+IfYGQZgbr0jNRGZLbwFUSaCDpJlIAb3IABUUPj/HKwlP6
UDWyNTbXMdSedXaeRHpdUki33gbD8Q8CqVB5sDyR2nZiXGBuM9v6drQRCRrGalm+
eFTLWPajM53kCLnnNDYv4upiS+2jqKHtQ/txyVqIDkNme5zCzxs63O8OJA57Go53
r4ULfSU0pUDo8c8K58JBXLeovIsXUkdC+KlBXVMNEd/HxUhGIsTyZcs2a/j/2mrr
gkzE2wAY2H5Q0rjMp0OCjomYUgG0/fNL5JZP11m3N84Au7H6qGwXoQbSe9YwK3S3
hZW64Grr6Twi5cB3+fe8QaWuwZt8lHbyySF9CSsTSI2wtAcmllNv4+BYu3foF9Z2
YlZJ8nPZkuajkhMnLIabaps3bsPhSlxD17/PizDOE0WM7GxazoJfNCUBIQARAQAB
tCRteCBtc3RybW5uIDxyaWNoYXJkQGdpbm51bmdhZ2FwLm9yZz6JAk4EEwEIADgW
IQRYiB2ee4h0BkApXNhM0u7DY+XTSAUCWIMpKQIbAwULCQgHAgYVCAkKCwIEFgID
AQIeAQIXgAAKCRBM0u7DY+XTSMJeEACYk6slxE1ktg1fR2Nrv/NU6zvPWUS28BVX
FplxYFTNEKZIS7I2ztL6BRnxtqoJoiMhTbzSINGW7lMk13r8D7r/VjtorcDVY6Fa
Tmn2b1SemmQVnlAaW1rQr1Zf+kyuShROrUbQMMibhV/Im5f6fkciFIjlgX4Ejyo0
ItuvOQUGXBfjXk78Q0j856u888PMj0QGOaiR1lZGu3Vv/Z0+JmwlYDIsVo7H4hPO
Nj+x0HWktjAMfoewMUbDi27k0UWU3pYDFcJ/xxgZQKVlk4iL7RpskSk9VvcW9Vns
MngEHDeovUxzx4/hz4B+O54CUmgYOBVhPoI+lBzKOejbeyGbb6qxOSjPoffjdhzo
hCYaxbIHSthzZ1o848mFYLaNnirYrbyWjGXNzBcuqdvXCic8nKAH9zNoImDEKsGF
Nem8iaClcL9vKVELRJ8VEffnTuooa3UPYBZfx53IyDvyrlKDQRH8pVEbr8ce3InB
XbNT19M8QiKggA+vI7h0YcBGMguhBQqSocGNzxE+EuX+xgsacR2GsrqwknhFLCHA
zZZEzxa7QsJtb1Meyo6JnXm0RvzKXYueGn4CM1LqGyXB25imf4TuDu6quZ7sDWu1
qLJ/I2qzXAC+ayDmJoNi52T1Oarw+HxtIFbQC8SrA9ArUfsZOBho6JahvdXPbFel
JZEDcWqyTrQvZWluc2llZGxlcmticyA8ZWluc2llZGxlcmtyZWJzQGdpbm51bmdh
Z2FwLm9yZz6JAk4EEwEIADgWIQRYiB2ee4h0BkApXNhM0u7DY+XTSAUCWIMqgwIb
AwULCQgHAgYVCAkKCwIEFgIDAQIeAQIXgAAKCRBM0u7DY+XTSDibEACJWr5TWA4u
6jlv6G1w4RZh7GR/rebd8D6/SpwAnyLRgYoicMRWMp/1QPSXMzTvLPk6SqmqwBSE
avD7v6FmjAuUwhWCPVSPE38qU3BjxMsZaEMLhSPQRuL0HFGO9WUE2rGSlQyuXMPF
inGAnaHgDoxtG0mtsVWvm3kgZ7bn1lQxwncfPu4fr8kXJSLXM3vhZrqwpze/F0Q5
q8YpVeLZPiK7LzxIDmiEt9FY7Z5ZIu3GCDu/uR8+GEDlnD2AoL7JnOK9ECxEXpUy
9cvr8ubOpocrshrqpIYWZLF1A+8KCUrMfTgPOCe5a8djrZGpelrlyVQkoW2W6L/3
D7Yj5j4H1c32J+OBc/VjASKHIwsIjaeyqttKtcv4TD3513YworWrSZskEQWzgtAK
6Rhr9FHMpw5ySxNgEMpIfwDyz8WrY5ymamDRVB1ZP898xJ2KlCd2HnN7NgKCylGO
jQiUN3je8PiSk6qERCw62lS/hFyxXDypVNMzHYenWyJm9vdd1md0VZGxM12UONeQ
DhkYF6sjOcN9D/2NOdQ2ySi664dBUlC+MfOOM5DTAxAYDyf6/eDMkZ470+ko5J2G
iG7Oj5mcHaAJH9pz+/SDCDf2XZzIIKKASOolpjKtP5QbICkK6cvZxZ5fxqKBQPka
neeol2r+5q7Nsn3Bsk0tq0tGzK7eDN0Qh7kCDQRYgykpARAA1hiPo3fYTLXMv8te
UJkyDXQjr+czb59yvHXWeRMjcp3PkgLCOPzI1Hj4ESYT6uCrcaiAy3bw4TUpHHKE
05+s/a9T23/OR46+gzkDbC0zHPGqvSY2vMMQpbgo5MAwOQLBvn65q5gH+yI9M6ZN
MHMHondUr8Xdsbv/CALMEDSNam3pgssLAL09Mo0b/vyC+uIDJxXNePOboYFGoZbH
VUrL1cmaW0OLVQbiyTK/dBZFPmY0ertSue7UqvyrCSgYCVnCEiF6fYpUFS9jeKO2
bP7Q23Kjie+IfeDP4GWJI8AqaEwOL1B3n5OqUA4vQrH4pIPnJXPkDE2j3969xiq5
d5tbjy1XrgyXE9vICJ8OghtIZPEf1z6j+N/296jnjFt/xfUZr57Y5VX3mLERbINH
pE25HF3RVY3wx8QfTjR4akHHK0JPiu/u8EiDcRL8HhC0ow5MMTQclG5I8OxrQi/R
7T06VeXonncYHFWqLClbasuL8cqGwaORVs2q+5wXIVQcEEcP5EkIJNfckVL9ihyy
uQj/q8+t+saGUVO8xPfnM/Aqn+XSnZbwkNk9TozEgGHl0gSP/WjRaWxHxeBEfGJ3
PoQDpyUYFFt/Y3zCicO+Seo9OD1pYkB2fQzyDxbwvcZr39kXboXNpP68qrLgcarg
aWMXcl0hpRUFzb9Ws1DnRCZmkUEAEQEAAYkCNgQYAQgAIBYhBFiIHZ57iHQGQClc
2EzS7sNj5dNIBQJYgykpAhsMAAoJEEzS7sNj5dNIGeMQAJOCa6hgeEUdZX+2eAYQ
SE/RHnXESekjiZAizu83FZ+m1UdKzBXERnB/SC9pBMhmM3QnQg1Ni/tWLSVA83gN
6osw5/gmLciYzQawcC91iWyksoPbsUeecku4S5GfDsSpJexXbn0oabWGBf7L73yh
shJ2DpbRohjZf/gTn1d7gyrB3h7WJIlCEbM/aHPe5a2hr8HDdm/YSxyyJI8ZN4mt
JbQ7pFXF29N9BtrCj7pOcog2rC72x68v1YJw7EVBhq+CeLZ/suRnVKhX17hlKgA3
uQFxPGfQtUlVNdwmgkTWJ8FaZzbXzpZPxa+0TnOQJpCLqWgFFYaaGytrQGuAmhvY
asSvunxkoq/nhu5rphSkJKuWYfMDE8Ux7rvpupkzSfGAraxNxBFVLQ8AYll81iD/
O7JNcHqupBFq7taPjLU6UwXYmQWptNx2pel6kokPhsV2TezM/oO5nMkgMot6hoaH
gZ7mw+0tWuWXcQW+GtWzMX7y/5vELleUOrUYsV76KJ3yY+u7LRm5nw0gH3Jjc1e0
TDQNviaUI6bbWPA2+bQhQMva3r7SyQH2BNdoijAMC+pQdJJseEHkLIrVDpaxQ+mH
cC2R3J27GAXJDagPilUARrOxD71gnFxJGiKAEvbtLORoxMsLl3uhosG3vyDL5nxy
+cbOghoUEZPyzUyfOFxMG3hy
=81oF
-----END PGP PUBLIC KEY BLOCK-----

**Pull request merged**